Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationItemDefault.h

```diff
@@ -55,7 +55,17 @@ NS_SWIFT_NAME(HorizontalNavigationItemDefault) IB_DESIGNABLE @interface BPKHoriz
  * @param name NSString the name for the navigation item
  * @param iconName BPKIconName the icon for the navigation item
  */
-- (instancetype)initWithName:(NSString *)name iconName:(BPKIconName)iconName NS_DESIGNATED_INITIALIZER;
+- (instancetype)initWithName:(NSString *)name iconName:(BPKIconName)iconName;
+
+/**
+ * Create a `BPKHorizontalNavigationItem` with a set of options and an optionaly selected option.
+ *
+ * @param name NSString the name for the navigation item
+ * @param iconName BPKIconName the icon for the navigation item
+ * @param showDot send YES to show a red notification dot
+ */
+- (instancetype)initWithName:(NSString *)name iconName:(BPKIconName)iconName showNotificationDot:(BOOL)showDot NS_DESIGNATED_INITIALIZER;
```

Comment:
I think the public interface should not have a method for initialising with an icon and dot if this is unsupported. It might still be good to have this method privately for the others to call?

Reply:
I think that's supported? What's not supported it usin git with a badge?

Reply:
Well, this class is [not declared](https://github.com/Skyscanner/backpack-ios/blob/master/Backpack/HorizontalNavigation/Classes/HorizontalNavigation.h) to be used directly so that's why I didn't think twice this initializer but we can still discuss if it's necessary anyway to do that change